AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -0,0 +1,194 @@
+    private let payment: Payment?
+    private let environment: Environment
+    private let supportedCardTypes: [CardType]
+    private let scope: String
+
+    private let maxCardsVisible = 4
+    private let publicBinLenght = 6
+    private let throttler = Throttler(minimumDelay: 0.5)
+
+    private var topCardTypes: [CardType] {
+        Array(supportedCardTypes.prefix(maxCardsVisible))
+    }
+
+    // MARK: Init view controller
+
+    internal init(configuration: CardComponent.Configuration,
+                  formStyle: FormComponentStyle,
+                  payment: Payment?,
+                  environment: Environment,
+                  supportedCardTypes: [CardType],
+                  scope: String) {
+        self.configuration = configuration
+        self.formStyle = formStyle
+        self.payment = payment
+        self.environment = environment
+        self.supportedCardTypes = supportedCardTypes
+        self.scope = scope
+        super.init(style: formStyle)
+    }
+
+    override internal func viewDidLoad() {
+        append(numberItem)
+        numberItem.showLogos(for: topCardTypes)
+
+        if configuration.showsSecurityCodeField {
+            let splitTextItem = FormSplitItem(items: [expiryDateItem, securityCodeItem], style: formStyle.textField)
+            append(splitTextItem)
+        } else {
+            append(expiryDateItem)
+        }
+
+        if configuration.showsHolderNameField {
+            append(holderNameItem)
+        }
+
+        if configuration.showsStorePaymentMethodField {
+            append(storeDetailsItem)
+        }
+
+        append(button.withPadding(padding: .init(top: 8, left: 0, bottom: -16, right: 0)))
+
+        super.viewDidLoad()
+    }
+
+    // MARK: Public methods
+
+    public weak var cardDelegate: CardViewControllerDelegate?
+
+    public var card: Card {
+        Card(number: numberItem.value,
+             securityCode: configuration.showsSecurityCodeField ? securityCodeItem.nonEmptyValue : nil,
+             expiryMonth: expiryDateItem.value[0...1],
+             expiryYear: "20" + expiryDateItem.value[2...3],
+             holder: configuration.showsHolderNameField ? holderNameItem.nonEmptyValue : nil)
+    }
+
+    public var storePayment: Bool {
+        configuration.showsStorePaymentMethodField ? storeDetailsItem.value : false
+    }
+
+    /// :nodoc:
+    public func stopLoading() {
+        button.showsActivityIndicator = false
+        view.isUserInteractionEnabled = true
+    }
+
+    /// :nodoc:
+    public func startLoading() {
+        button.showsActivityIndicator = true
+        view.isUserInteractionEnabled = false
+    }
+
+    public func update(binInfo: BinLookupResponse) {
+        self.securityCodeItem.update(cardBrands: binInfo.brands ?? [])
+
+        guard let brands = binInfo.brands else { return }
+        self.numberItem.showLogos(for: brands.isEmpty ? self.topCardTypes : brands.map(\.type))
+    }
```

Comment:
why are those public? the whole class is internal, right?

Reply:
Yep, that is a good point